java/clients/src/main/java/sleeper/clients/docker/stack/TableStack.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.clients.docker.stack;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.s3.AmazonS3;
+
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.core.statestore.StateStoreException;
+import sleeper.statestore.dynamodb.DynamoDBStateStore;
+import sleeper.statestore.dynamodb.DynamoDBStateStoreCreator;
+import sleeper.table.job.TableCreator;
+
+import static sleeper.clients.docker.Utils.tearDownBucket;
+import static sleeper.configuration.properties.table.TableProperty.ACTIVE_FILEINFO_TABLENAME;
+import static sleeper.configuration.properties.table.TableProperty.DATA_BUCKET;
+import static sleeper.configuration.properties.table.TableProperty.PARTITION_TABLENAME;
+import static sleeper.configuration.properties.table.TableProperty.READY_FOR_GC_FILEINFO_TABLENAME;
+
+public class TableStack implements DockerStack {
+    private final InstanceProperties instanceProperties;
+    private final TableProperties tableProperties;
+    private final AmazonS3 s3Client;
+    private final AmazonDynamoDB dynamoDB;
+
+    private TableStack(Builder builder) {
+        instanceProperties = builder.instanceProperties;
+        tableProperties = builder.tableProperties;
+        s3Client = builder.s3Client;
+        dynamoDB = builder.dynamoDB;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static TableStack from(InstanceProperties instanceProperties, TableProperties tableProperties,
+                                  AmazonS3 s3Client, AmazonDynamoDB dynamoDB) {
+        return builder().instanceProperties(instanceProperties)
+                .tableProperties(tableProperties)
+                .s3Client(s3Client).dynamoDB(dynamoDB)
+                .build();
+    }
+
+    public void deploy() {
+        s3Client.createBucket(tableProperties.get(DATA_BUCKET));
+
+        new TableCreator(s3Client, dynamoDB, instanceProperties).createTable(tableProperties);
+        try {
+            DynamoDBStateStore stateStore = new DynamoDBStateStoreCreator(instanceProperties, tableProperties, dynamoDB).create();
+            stateStore.initialise();
```

Comment:
The TableCreator creates the state store, although it doesn't initialise it. You can use StateStoreFactory instead of DynamoDBStateStoreCreator to get an instance of StateStore to call initialise.